firmware/fw_base.S

```diff
@@ -13,6 +13,11 @@
 #include <sbi/sbi_scratch.h>
 #include <sbi/sbi_trap.h>
 
+	.align 3
+	.section .data, "aw"
+_boot_hart_done:
+	RISCV_PTR	0
+
```

Comment:
It's good to have _boot_hart_done in data section but let's try to keep _start as first symbol in fw_base.S for readability.

I would suggest to have _boot_hart_done just before _hartid_to_scratch.

Reply:
done